From e085738e5ad571d0caa2088664c305e79fe7bc37 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Mon, 10 Jun 2019 16:29:04 +0800
Subject: [PATCH] kexec: dcss-pll: enable pll output and clear PD down bit to
 avoid kernel hang

There is a dtg module in DCSS subsystem. Kexec kernel always hang when
accessing dtg registers. This is because that the PLL and powerdomain
for DCSS is disabled by dcss_runtime_suspend in the first kernel. So we
enable clk output and power up VIDEO_PLL2 by calling dcss_pll_enable for
the kexec kernel to avoid kernel hang.

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-common.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/imx/dcss/dcss-common.c b/drivers/gpu/imx/dcss/dcss-common.c
index 704bf4bfeb49..cf9384d3ff0a 100644
--- a/drivers/gpu/imx/dcss/dcss-common.c
+++ b/drivers/gpu/imx/dcss/dcss-common.c
@@ -294,6 +294,12 @@ static int dcss_clks_init(struct dcss_soc *dcss)
 
 	dcss->clks_on = true;
 
+	/* We must make sure that VIDEO_PLL2 enable clk output and clear power
+	 * down bit when finishing enabling clks, just as what dcss_runtime_resume
+	 * does, if not, kexec kernel will hang when accessing dcss dtg registers.
+	 */
+	dcss_pll_enable(dcss);
+
 	return 0;
 
 err:
-- 
2.17.1

